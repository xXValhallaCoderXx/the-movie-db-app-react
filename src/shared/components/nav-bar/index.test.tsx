import React from "react";
import {shallow, mount} from "enzyme";
import {MemoryRouter} from "react-router-dom";
import NavBar from "./index";

describe("Nav Bar Component", () => {
  it("should render correctly", () => {
    const component = shallow(<NavBar movieID="" />);
    expect(component).toMatchSnapshot();
  });
  it("includes link to Home page and Mobile view link is not showing", () => {
    const wrapper = mount(
      <MemoryRouter>
        <NavBar movieID="" />
      </MemoryRouter>
    );
    expect(
      wrapper
        .find("#home-link")
        .at(0)
        .prop("to")
    ).toEqual("/movies");

    expect(wrapper.find("#mobile-back").exists()).toBeFalsy();
  });
  it("should have a back link on mobile if movie is selected", () => {
    const wrapper = mount(
      <MemoryRouter>
        <NavBar mobile movieID="1234" />
      </MemoryRouter>
    );
    expect(
      wrapper
        .find("#mobile-back")
        .at(0)
        .prop("to")
    ).toEqual("/movies");
  });
});
