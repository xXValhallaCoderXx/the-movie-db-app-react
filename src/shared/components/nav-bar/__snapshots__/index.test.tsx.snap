// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Bar Component should render correctly 1`] = `
<nav
  className="flex items-center justify-between flex-wrap bg-tmd-dark p-4 fixed w-full"
>
  <div
    className="flex items-center flex-shrink-0 text-white mr-6"
  >
    <svg
      className="fill-current h-8 w-8 mr-2"
      height="54"
      viewBox="0 0 54 54"
      width="54"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M13.5 22.1c1.8-7.2 6.3-10.8 13.5-10.8 10.8 0 12.15 8.1 17.55 9.45 3.6.9 6.75-.45 9.45-4.05-1.8 7.2-6.3 10.8-13.5 10.8-10.8 0-12.15-8.1-17.55-9.45-3.6-.9-6.75.45-9.45 4.05zM0 38.3c1.8-7.2 6.3-10.8 13.5-10.8 10.8 0 12.15 8.1 17.55 9.45 3.6.9 6.75-.45 9.45-4.05-1.8 7.2-6.3 10.8-13.5 10.8-10.8 0-12.15-8.1-17.55-9.45-3.6-.9-6.75.45-9.45 4.05z"
      />
    </svg>
    <Link
      id="home-link"
      to="/movies"
    >
      <span
        className="font-semibold text-xl tracking-tight text-tmd-green hover:text-tmd-light cursor-pointer"
      >
        The Movie DB App
      </span>
    </Link>
  </div>
</nav>
`;
